client/console/src/main/java/org/apache/syncope/client/console/panels/HistoryConfDetails.java

```diff
@@ -0,0 +1,259 @@
+            }
+        });
+        this.availableHistoryConfTOs = availableHistoryConfTOs;
+        this.currentHistoryConfTO = currentHistoryConfTO;
+
+        Form form = initDropdownDiffConfForm();
+        add(form);
+        if (availableHistoryConfTOs.isEmpty()) {
+            form.setVisible(false);
+        } else {
+            form.setVisible(true);
+        }
+
+        initSingleContent();
+    }
+
+    private void showConfigurationSinglePanel() {
+        final ConsoleLayoutInfo info = getJSONInfo(currentHistoryConfTO);
+
+        jsonEditorPanel = new JsonEditorPanel(null, new PropertyModel<String>(info, "content"), Boolean.TRUE, null) {
+
+            private static final long serialVersionUID = -8927036362466990179L;
+
+            @Override
+            public void onSubmit(final AjaxRequestTarget target, final Form<?> form) {
+                try {
+                    modal.close(target);
+                } catch (Exception e) {
+                    LOG.error("While updating console layout info for history configuration {}", info.getKey(), e);
+                    SyncopeConsoleSession.get().error(StringUtils.isBlank(e.getMessage())
+                            ? e.getClass().getName() : e.getMessage());
+                }
+                ((BasePage) pageRef.getPage()).getNotificationPanel().refresh(target);
+            }
+        };
+        jsonEditorPanel.setOutputMarkupId(Boolean.TRUE);
+
+        addOrReplace(jsonEditorPanel);
+    }
+
+    private void initSingleContent() {
+        showConfigurationSinglePanel();
+    }
+
+    private void initMultipleContent(final List<T> historyConfTOs) {
+        showConfigurationDiffPanel(historyConfTOs);
+    }
+
+    private void showConfigurationDiffPanel(final List<T> historyConfTOs) {
+        final List<ConsoleLayoutInfo> infos = new ArrayList<>();
+        for (T historyConfTO : historyConfTOs) {
+            infos.add(getJSONInfo(historyConfTO));
+        }
+
+        jsonEditorPanel = new JsonDiffEditorPanel(null, new PropertyModel<String>(infos.get(0), "content"),
+                new PropertyModel<String>(infos.get(1), "content"), null) {
+
+            private static final long serialVersionUID = -8927036362466990179L;
+
+            @Override
+            public void onSubmit(final AjaxRequestTarget target, final Form<?> form) {
+                try {
+                    modal.close(target);
+                } catch (Exception e) {
+                    LOG.error("While updating console layout info for history configurations {}", infos, e);
+                    SyncopeConsoleSession.get().error(StringUtils.isBlank(e.getMessage())
+                            ? e.getClass().getName() : e.getMessage());
+                }
+                ((BasePage) pageRef.getPage()).getNotificationPanel().refresh(target);
+            }
+        };
+
+        replace(jsonEditorPanel);
+    }
+
+    private ConsoleLayoutInfo getJSONInfo(final T historyConfTO) {
```

Comment:
`ConsoleLayoutInfo` has a completely different purpose, please define your own class (if needed, wouldn't `Pair` fit for the job anyway?)